java-client/src/main/java/com/cumulocity/sdk/client/ResponseParser.java

```diff
@@ -77,14 +80,32 @@ public void checkStatus(Response response, int... expectedStatusCodes) throws SD
 
     protected String getErrorMessage(Response response, int status) {
         String errorMessage = "Http status code: " + status;
+        String responseHeader = null;
 
         if (response.hasEntity()) {
             String errorRepresentation = getErrorRepresentation(response);
             if (errorRepresentation == null) {
                 errorRepresentation = NO_ERROR_REPRESENTATION;
             }
+            Map<String, List<String>> responseHeadermap = response.getStringHeaders();
+            if (responseHeadermap != null) {
+                for (Map.Entry<String, List<String>> entry : responseHeadermap.entrySet()) {
+                    if (entry.getKey().equals("Forwarded")
+                            || entry.getKey().equals("Location")
+                            || entry.getKey().equals("Via")) {
+                        responseHeader += entry.getKey() +
```

Comment:
Given that `responseHeader` is initialized as null, _I think_ the `+=` operator will append the text to null. I.e. the text will be prefixed with null.

Probably we should just initialize the variable with an empty string instead of null.